sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -162,9 +168,20 @@ def _transform(self, X, fitting):
 
         # collect all the possible feature names and build sparse matrix at
         # same time
+        if self.drop_first:
+            to_drop = {}
         for x in X:
             for f, v in six.iteritems(x):
                 if isinstance(v, six.string_types):
+                    if self.drop_first and f not in to_drop:
```

Comment:
Yes, I agree. It also results in more readable code. I'm going to update the PR today, I'm making the final checks.